test/py/TestAsyncioClient.py

```diff
@@ -0,0 +1,350 @@
+        print('testTypedef')
+        x = 0xffffffffffffff  # 7 bytes of 0xff
+        y = yield from self.client.testTypedef(x)
+        self.assertEqual(y, x)
+
+    @async_test
+    def testMapMap(self):
+        print('testMapMap')
+        x = {
+            -4: {-4: -4, -3: -3, -2: -2, -1: -1},
+            4: {4: 4, 3: 3, 2: 2, 1: 1},
+        }
+        y = yield from self.client.testMapMap(42)
+        self.assertEqual(y, x)
+
+    @async_test
+    def testMulti(self):
+        print('testMulti')
+        xpected = Xtruct(string_thing='Hello2', byte_thing=74, i32_thing=0xff00ff, i64_thing=0xffffffffd0d0)
+        y = yield from self.client.testMulti(xpected.byte_thing,
+                                             xpected.i32_thing,
+                                             xpected.i64_thing,
+                                             {0: 'abc'},
+                                             Numberz.FIVE,
+                                             0xf0f0f0)
+        self.assertEqual(y, xpected)
+
+    @async_test
+    def testException(self):
+        print('testException')
+        yield from self.client.testException('Safe')
+        try:
+            yield from self.client.testException('Xception')
+            self.fail("should have gotten exception")
+        except Xception as x:
+            self.assertEqual(x.errorCode, 1001)
+            self.assertEqual(x.message, 'Xception')
+            # TODO ensure same behavior for repr within generated python variants
+            # ensure exception's repr method works
+            # x_repr = repr(x)
+            # self.assertEqual(x_repr, 'Xception(errorCode=1001, message=\'Xception\')')
+
+        try:
+            yield from self.client.testException('TException')
+            self.fail("should have gotten exception")
+        except TException as x:
+            pass
+
+        # Should not throw
+        yield from self.client.testException('success')
+
+    @async_test
+    def testMultiException(self):
+        print('testMultiException')
+        try:
+            yield from self.client.testMultiException('Xception', 'ignore')
+        except Xception as ex:
+            self.assertEqual(ex.errorCode, 1001)
+            self.assertEqual(ex.message, 'This is an Xception')
+
+        try:
+            yield from self.client.testMultiException('Xception2', 'ignore')
+        except Xception2 as ex:
+            self.assertEqual(ex.errorCode, 2002)
+            self.assertEqual(ex.struct_thing.string_thing, 'This is an Xception2')
+
+        y = yield from self.client.testMultiException('success', 'foobar')
+        self.assertEqual(y.string_thing, 'foobar')
+
+    @async_test
+    def testOneway(self):
+        print('testOneway')
+        start = time.time()
+        yield from self.client.testOneway(1)  # type is int, not float
+        end = time.time()
+        self.assertTrue(end - start < 3,
+                        "oneway sleep took %f sec" % (end - start))
```

Comment:
It seems rather an issue of original test code, but 3 seconds are way too large for this test to make any sense.
Something like 200 ms should be enough.